controllers/argocd/service.go

```diff
@@ -461,6 +448,27 @@ func (r *ReconcileArgoCD) reconcileServerService(cr *argoproj.ArgoCD) error {
 	if err := controllerutil.SetControllerReference(cr, svc, r.Scheme); err != nil {
 		return err
 	}
+
+	existingSVC := &corev1.Service{}
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, svc.Name, existingSVC) {
+		if !cr.Spec.Server.IsEnabled() {
+			return r.Client.Delete(context.TODO(), svc)
+		}
+		if ensureAutoTLSAnnotation(r.Client, svc, common.ArgoCDServerTLSSecretName, cr.Spec.Server.WantsAutoTLS()) {
+			return r.Client.Update(context.TODO(), svc)
+		}
+		if !cr.Spec.Repo.IsEnabled() {
+			return nil
+		}
```

Comment:
```suggestion
		if !cr.Spec.Server.IsEnabled() {
			return nil
		}
```
Seems like there is an existing bug in this check. It should be `.Server` and not `.Repo`. This check is used to skip service creation for the server is it is disabled by setting `cr.spec.server.enabled: true` in `ArgoCD` CR. Let's also move this check out of update loop towards the end i.e before `Create` so that it correctly prevents the creation of service if  server is disabled.